src/CommandDialog.h

```diff
@@ -0,0 +1,53 @@
+#ifndef CONTROLDIALOG_OPENMSX_H
+#define CONTROLDIALOG_OPENMSX_H
+
+#include "ui_CommandDialog.h"
+#include <QDialog>
+
+
+struct CommandRef
+{
+    QString name;
+    QString description;
+    QString source;
+    QString icon;
+    int index;
+
+    bool operator<(const CommandRef& ref) const {
+        return index < ref.index;
+    }
+};
+
+
+class CommandDialog : public QDialog, private Ui::CommandDialog
+{
+    Q_OBJECT
+public:
+    CommandDialog(QList<CommandRef>& commands, QWidget* parent = nullptr);
+
+    QString icon() const;
+    QString name() const;
+    QString command() const;
+    QString function() const;
```

Comment:
I was going to say: these should probably return `const QString&`.
But then later I did not find any implementation of these 4 methods. Should these just be removed?